multi-transfer-esdt/src/lib.rs

```diff
@@ -68,8 +72,16 @@ pub trait MultiTransferEsdt:
             // emit event before the actual transfer so we don't have to save the tx_nonces as well
             self.transfer_performed_event(batch_id, eth_tx.tx_nonce);
 
-            valid_dest_addresses_list.push(eth_tx.to);
-            valid_payments_list.push(EsdtTokenPayment::new(eth_tx.token_id, 0, eth_tx.amount));
+            if self.blockchain().is_smart_contract(&eth_tx.to.clone()) {
+                let _: IgnoreValue = self
+                    .get_bridge_proxy_contract_proxy_instance()
+                    .deposit(&eth_tx)
+                    .with_esdt_transfer((eth_tx.token_id.clone(), 0, eth_tx.amount.clone()))
+                    .execute_on_dest_context();
+            } else {
+                valid_dest_addresses_list.push(eth_tx.to);
+                valid_payments_list.push(EsdtTokenPayment::new(eth_tx.token_id, 0, eth_tx.amount));
+            }
         }
 
         let payments_after_wrapping = self.wrap_tokens(valid_payments_list);
```

Comment:
I think we should first try to wrap tokens, no matter if its a SC-call or not and only then to make a deposit on bridge proxy.

Reply:
agreed